### PR TITLE
Linter for powershell syntax errors

### PR DESCRIPTION
**Pull request by zigford**
* Moved the RunPowershell function out of the psscriptanalyzer into autoload/ale/powershell.vim so that it can be used by other powershell things.
* Added powershell linter which uses powershell itself to parse a script and output any syntax errors
* Added documentation
* Added tests.

### CODE REVIEW

**Comment by zigford** [comment]
I think I'm failing because of the custom linting rules and those rules are conflicting with each other with how the code is currently factored. Should I refactor for the linting rules?

File: ale_linters/powershell/powershell.vim [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                " matches
+                call add(l:output, l:item)
+                let l:matchcount = 1
+            endif
+            let l:item = {
+            \   'lnum': str2nr(l:match[1]),
+            \   'col': str2nr(l:match[2]),
+            \   'type': 'E',
+            \}
+
```
Comment (w0rp) [inline ale_linters/powershell/powershell.vim]: Remove this blank line.

File: ale_linters/powershell/powershell.vim [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            " First match consists of 2 capture groups, and
+            " can capture the line and col
+            if exists('l:item')
+                " We may be here because the last syntax
+                " didn't emit a code, and so only had 2
+                " matches
+                call add(l:output, l:item)
+                let l:matchcount = 1
+            endif
+            let l:item = {
```
Comment (w0rp) [inline ale_linters/powershell/powershell.vim]: Add blank line after `endif`.

File: ale_linters/powershell/powershell.vim [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            " append the code we now know
+            call add(l:output, l:item)
+            unlet l:item
+
+            if len(l:match[1]) > 0
+                for l:i in l:output
+                    let l:i['code'] = l:match[1]
+                endfor
+            endif
+            " Reset the matchcount so we can begin gathering
```
Comment (w0rp) [inline ale_linters/powershell/powershell.vim]: Add a blank line after `endif`.

File: ale_linters/powershell/powershell.vim [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            " matches for the next syntax error
+            let l:matchcount = 0
+
```
Comment (w0rp) [inline ale_linters/powershell/powershell.vim]: Remove this blank line.

**Comment by zigford** [comment]
Apologies for causing all these build failures. So that I don't cause any more in future PR's, could you explain again what is going wrong now?

**Comment by w0rp** [comment]
The remaining issue is that the Vim help file's list of supported tools needs to match the Markdown list.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**